### PR TITLE
KBI 0009:  ingesting data into a dataset clone

### PR DESCRIPTION
**Pull request by adswa**
This takes some of the notes created by @mih in #12 and distills them into a knowledge base item on re-ingesting data into a dataset clone.

### CODE REVIEW

File: kbi/0009/index.rst [hunk 1]
```diff
@@ -0,0 +1,99 @@
+.. index::
+   single: <topic>; <subtopic>
+
+.. highlight:: console
+
+KBI0009: How to re-ingest file content in a dataset clone
+=========================================================
+
+:authors: Adina Wagner <adina.wagner@t-online.de>
+:discussion: <link>
```
Comment (adswa) [inline kbi/0009/index.rst:10]:
```suggestion
:discussion: https://github.com/psychoinformatics-de/knowledge-base/pull/37
```

File: kbi/0009/index.rst [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+provide:
+To obtain a copy of the data in a dataset in location A in a different
+location B, one can clone the dataset from A to B and retrieve its contents.
+
+This KBI, though, outlines a rather unusual different method:
+Given that the dataset in A contains lots of data, and given that a complete
+**copy of the data already exists outside of a DataLad dataset in location B**,
+a plain clone from the dataset in A to B can be used to re-ingest the data.
+The result is identical to having cloned and retrieved data, but without
+duplicating the data transfer and instead using the pre-existing copy of the data.
```
Comment (mih) [inline kbi/0009/index.rst:26]:
```suggestion
If a dataset at location A contains lots of data, and given that a complete
**copy of the data already exists outside of a DataLad dataset in location B**,
a plain clone from the dataset in A to B can be used to re-ingest the data.
The result is identical to having cloned and retrieved data, but without
duplicating the data transfer and instead using the pre-existing copy of the data.
```

File: kbi/0009/index.rst [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+
+Example workflow
+----------------
+
+Lets set up a toy example dataset at location A::
```
Comment (mih) [inline kbi/0009/index.rst:32]:
```suggestion
Let's set up a toy example dataset at location A::
```

File: kbi/0009/index.rst [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+    # on one system
+    $ datalad create location-A
+    create(ok): /tmp/location-A (dataset)
+
+Let's put some files inside. In this toy example its only a single small file,
```
Comment (mih) [inline kbi/0009/index.rst:38]:
```suggestion
Let's put some files inside. In this toy example it is only a single small file,
```

File: kbi/0009/index.rst [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    $ ls -l .
+    lrwxrwxrwx 1 adina adina 130 Apr 21 12:51 big.dat -> .git/annex/objects/1G/4J/MD5E-s10485760--49c360327aabc60e0b75e9bff4bee060.dat/MD5E-s10485760--49c360327aabc60e0b75e9bff4bee060.dat
+
+Pay attention what the current commit SHA is to later easily identify any data
+changes::
+
+   $ git log -1 --pretty=oneline | cat
+   ba083bf39f87bc97c60d70b1ff0c69e41cbf28a2 [DATALAD RUNCMD] dd if=/dev/urandom bs=1M count=10 of=/ho...
+
+We can now copy or move over the files from a pre-existing copy on location-B into the worktree of the dataset::
```
Comment (mih) [inline kbi/0009/index.rst:70]:
```suggestion
We can now copy or move over the files from a pre-existing copy into the worktree of the dataset::
```

File: kbi/0009/index.rst [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    $ git log -1 --pretty=oneline | cat
+    ba083bf39f87bc97c60d70b1ff0c69e41cbf28a2 [DATALAD RUNCMD] dd if=/dev/urandom bs=1M count=10 of=/ho...
+
+If the commit Shasum differs from the previously seen Shasum now, though, the
+data that has been ingested isn't identical to the data originally tracked, for
+example due to accidental corruption.
+
+
+
+
```
Comment (mih) [inline kbi/0009/index.rst:99]:
```suggestion
If the commit SHA differs from the previously seen SHA now, though, the
data that has been ingested is not identical to the data originally tracked, for
example due to accidental corruption.
```

Align with label used above.

**Review by mih — APPROVED** [review]
Works for me. Thx!

**Pull request merged**